react/features/mobile/external-api/middleware.ts

```diff
@@ -186,18 +193,45 @@ externalAPIEnabled && MiddlewareRegistry.register(store => next => action => {
     }
 
     case ENDPOINT_MESSAGE_RECEIVED: {
-        const { participant, data } = action;
+        const { participant, data: json } = action;
+        const { text } = json.transcript[0];
+        const STABLE_TRANSCRIPTION_FACTOR = 0.85;
 
-        if (data?.name === ENDPOINT_TEXT_MESSAGE_NAME) {
+        if (json?.name === ENDPOINT_TEXT_MESSAGE_NAME) {
             sendEvent(
                 store,
                 ENDPOINT_TEXT_MESSAGE_RECEIVED,
                 /* data */ {
-                    message: data.text,
+                    message: json.text,
                     senderId: participant.getId()
                 });
         }
 
+        const { skipInterimTranscriptions }
```

Comment:
Let's not duplicate this logic here. Reuse the one web is using please. In addition, you are not checking the message type here. The web code is already doing that check.